srcCpp/Infrastructure_pro.cxx

```diff
@@ -580,10 +581,14 @@ bool configureShmemTransport(
     RTIBool success = RTI_FALSE;
     int retcode;
     int key = rand();
-    int minBufferSize = parentMsgSizeMax;
+    int minBufferSize = 1048576;
     int step = 1048576; // 1MB
     int maxBufferSize = (std::max)(60817408 /* 58MB */, parentMsgSizeMax);
 
+    if (!messageSizeMaxSet) {
+
```

Comment:
I am guessing this is in the wrong place, right? We are missing after the do-while a check with something like:

If size greater than 64k and I couldn't reserve memory for doing synchronous, default all the settings and try enabling asynchronous. Right?

Reply:
In fact, that if had to be removed all together. Below we will adjust the transport size if messageSizeMaxSet is not true